pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -998,94 +992,80 @@ impl DeltaLayerInner {
         reconstruct_state: &mut ValuesReconstructState,
         ctx: &RequestContext,
     ) {
-        let vectored_blob_reader = VectoredBlobReader::new(&self.file);
-        let mut ignore_key_with_err = None;
-
         let max_vectored_read_bytes = self
             .max_vectored_read_bytes
             .expect("Layer is loaded with max vectored bytes config")
             .0
             .into();
         let buf_size = Self::get_min_read_buffer_size(&reads, max_vectored_read_bytes);
-        let mut buf = Some(IoBufferMut::with_capacity(buf_size));
+
+        let mut ignore_key_with_err = None;
```

Comment:
This value is `Copy`d into the spawned IO, but the code inside the spawned IO makes me think this was meant as a cancellation / "don't do more work because we failed an earlier IO" type thing.

Reply:
It was an optimisation to avoid doing extra work if we already failed a key. Looking at it now, it's of dubious value.
We save some CPU work, but that's it. We can either remove it altogether or make it more clear that it's per IO future by moving the initialization into the `for read ...` loop.

Reply:
I'll remove this.

Reply:
I think the `ios.is_empty()` assert that we added would fire